sdk/notificationhubs/notification-hubs/samples-dev/sendDirectNotificationBatch.ts

```
@@ -0,0 +1,95 @@
+import { delay } from "@azure/core-util";
+import { getNotificationOutcomeDetails } from "@azure/notification-hubs/client/getNotificationOutcomeDetails";
+import { sendNotification } from "@azure/notification-hubs/client/sendNotification";
+
+// Load the .env file if it exists
+dotenv.config();
+
+// Define connection string and hub name
+const connectionString = process.env.NOTIFICATIONHUBS_CONNECTION_STRING || "<connection string>";
+const hubName = process.env.NOTIFICATION_HUB_NAME || "<hub name>";
+
+// Define message constants
+const DUMMY_DEVICE = "00fc13adff785122b4ad28809a3420982341241421348097878e577c991de8f0";
+const deviceHandle = process.env.APNS_DEVICE_TOKENS?.split(",") || [DUMMY_DEVICE];
+
+async function main() {
+  const context = createClientContext(connectionString, hubName);
+
+  const messageBody = `{ "aps" : { "alert" : { title: "Hello", body: "Hello there SDK Review!" } } }`;
+
+  const notification = createAppleNotification({
+    body: messageBody,
+    headers: {
+      "apns-priority": "10",
+      "apns-push-type": "alert",
+    },
+  });
+
+  const result = await sendNotification(context, notification, { deviceHandle });
+
+  console.log(`Direct send Tracking ID: ${result.trackingId}`);
+  console.log(`Direct send Correlation ID: ${result.correlationId}`);
+
+  // Only available in Standard SKU and above
+  if (result.notificationId) {
+    console.log(`Direct send Notification ID: ${result.notificationId}`);
+
+    const results = await getNotificationDetails(context, result.notificationId);
+    if (results) {
+      console.log(JSON.stringify(results, null, 2));
+    }
+  }
+}
+
+async function getNotificationDetails(
+  context: NotificationHubsClientContext,
+  notificationId: string
+): Promise<NotificationDetails | undefined> {
+  let state: NotificationOutcomeState = "Enqueued";
+  let count = 0;
+  let result: NotificationDetails | undefined;
```

to avoid the potential `undefined` return, maybe you can cache the last error and throw if result is still undefined at the end? It feels weird to have the sample use `undefined` as a return value when `getNotificationOutcomeDetails` itself does not

---

Moving the logic so as to not throw on a 404, but throw on the rest.